tests/test_nhg.py

```diff
@@ -384,20 +799,891 @@ def asic_route_nhg_fvs(k):
+
+            nexthop = ','.join(nexthop)
+            ifname = ','.join(ifname)
+            fvs = swsscommon.FieldValuePairs([("nexthop", nexthop), ("ifname", ifname)])
+
+            return fvs
+
+        def gen_nhg_index(nhg_number):
+            return "group{}".format(nhg_number)
+
+        def gen_valid_binary():
+            nonlocal r
+
+            while True:
+                r += 1
+                binary = fmt.format(r)
+                # We need at least 2 ports for a nexthop group
+                if binary.count('1') <= 1:
+                    continue
+                return binary
+
+        def create_temp_nhg():
+            nonlocal nhg_count
+
+            binary = gen_valid_binary()
+            nhg_fvs = gen_nhg_fvs(binary)
+            nhg_index = gen_nhg_index(nhg_count)
+            nhg_ps.set(nhg_index, nhg_fvs)
+            nhg_count += 1
+
+            return nhg_index, binary
+
+        def delete_nhg():
+            nonlocal first_valid_nhg
+
+            del_nhg_index = gen_nhg_index(first_valid_nhg)
+            del_nhg_id = asic_nhgs[del_nhg_index]
+
+            nhg_ps._del(del_nhg_index)
+            asic_nhgs.pop(del_nhg_index)
+            first_valid_nhg += 1
+
+            return del_nhg_id
+
+        def nhg_exists(nhg_index):
+            return self.get_nhg_id(nhg_index, dvs) is not None
+
+        # Create interfaces
+        for i in range(MAX_PORT_COUNT):
+            self.config_intf(i, dvs)
+
+        asic_nhgs = {}
+
+        # Add first batch of next hop groups to reach the NHG limit
+        while nhg_count < MAX_ECMP_COUNT:
+            r += 1
+            binary = fmt.format(r)
+            # We need at least 2 ports for a nexthop group
+            if binary.count('1') <= 1:
+                continue
+            nhg_fvs = gen_nhg_fvs(binary)
+            nhg_index = gen_nhg_index(nhg_count)
+            nhg_ps.set(nhg_index, nhg_fvs)
+
+            # Wait for the group to be added
+            asic_db.wait_for_n_keys(self.ASIC_NHG_STR, asic_nhgs_count + nhg_count + 1)
+
+            # Save the NHG index/ID pair
+            asic_nhgs[nhg_index] = self.get_nhg_id(nhg_index, dvs)
+
+            nhg_count += 1
+
+        # Add a new next hop group - it should create a temporary one instead
+        prev_nhgs = asic_db.get_keys(self.ASIC_NHG_STR)
+        nhg_index, _ = create_temp_nhg()
+
+        # Assert no new group has been added
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, MAX_ECMP_COUNT)
+
+        # Assert the same NHGs are in ASIC DB
+        assert prev_nhgs == asic_db.get_keys(self.ASIC_NHG_STR)
+
+        # Delete an existing next hop group
+        del_nhg_id = delete_nhg()
+
+        # Wait for the key to be deleted
+        asic_db.wait_for_deleted_entry(self.ASIC_NHG_STR, del_nhg_id)
+
+        # Wait for the temporary group to be promoted and replace the deleted
+        # NHG
+        asic_db.wait_for_n_keys(self.ASIC_NHG_STR, MAX_ECMP_COUNT)
+
```

Comment:
can you assert SAI NHOP ID will change here.

Reply:
Done.